### PR TITLE
feat: funding flow changes (WIP)

### PR DESCRIPTION
**Pull request by steliosrammos**

### CODE REVIEW

File: src/modules/profile/pages/profileSettings/views/ProfileSettingsNotifications.tsx [hunk 1]
```diff
@@ -43,14 +25,10 @@ export const ProfileSettingsNotifications = () => {
...
-          <CreatorNotificationsSkeleton />
-        ) : (
-          <CreatorNotifications creatorNotificationSettings={creatorNotificationSettings} />
-        )}
+        <CreatorNotifications />
         <Divider />
-        {userNotificationSettings && <UserNotifications userNotificationSettings={userNotificationSettings} />}
-        {!loading && <FollowedProjectsList />}
+        <UserNotifications />
+        <FollowedProjectsList />
```
Comment (Agill-Sheron) [inline src/modules/profile/pages/profileSettings/views/ProfileSettingsNotifications.tsx:31]: Much more cleaner like this too, well done!

File: src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx [hunk 2]
```diff
@@ -1,49 +1,202 @@
...
     setErrorstate,
   } = useFundingFormAtom()
 
-  if (!hasSelectedRewards) {
-    return null
+  /*
+   Set the email from the user to the funding form. We do this because the input field
+   is not shown if the user already has an email.
+  */
+  console.log('USER', user)
```
Comment (Agill-Sheron) [inline src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx:63]: console.log

File: src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx [hunk 3]
```diff
@@ -1,49 +1,202 @@
...
+  useEffect(() => {
+    if (user?.email) setTarget({ target: { name: 'email', value: user.email } })
+    else setTarget({ target: { name: 'email', value: '' } })
+  }, [user])
+
+  const [isEmailAvailable, { loading: userEmailIsAvailableLoading }] = useUserEmailIsAvailableLazyQuery()
+  console.log('email', email)
```
Comment (Agill-Sheron) [inline src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx:70]: console.log

File: src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx [hunk 4]
```diff
@@ -1,49 +1,202 @@
...
+              label="Receive direct creator emails. If you accept, you will receive updates directly from this project via email."
+              htmlFor="creator-email-toggle"
+            >
+              <Switch
+                id="creator-email-toggle"
+                isChecked={followProject}
+                onChange={(e) => {
+                  setTarget({ target: { name: 'followProject', value: e.target.checked } })
+                  console.log('Follow Project Toggle:', e.target.checked)
+                  console.log(
```
Comment (Agill-Sheron) [inline src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx:179]: console.log

**Review by Agill-Sheron — CHANGES_REQUESTED** [review]

File: src/modules/profile/pages/profileSettings/components/UserNotifications.tsx [hunk 5]
```diff
@@ -1,143 +1,90 @@
...
-                ConfigName.IS_ENABLED,
-                e.target.checked ? 'true' : 'false',
-              )
-            }
-          />
-        </HorizontalFormField>
-      </VStack>
-    </VStack>
+    <>
+      {loadingUserNotificationSettings && <Skeleton />}
```
[the remark this answers is not in the document]
Comment (steliosrammos) [reply src/modules/profile/pages/profileSettings/components/UserNotifications.tsx:27]: Not sure actually. Was there styling before?

File: src/modules/profile/pages/profileSettings/views/ProfileSettingsNotifications.tsx (hunk 1 above)
Comment (steliosrammos) [reply src/modules/profile/pages/profileSettings/views/ProfileSettingsNotifications.tsx:31]: I did this at first to re-use the hook in the funding flow, and have it update the notification setting real-time, but then I realised it wouldn't work with logged out users, so I had to scrap all that 🥲 but hey, we got a nice refactor out of it :p

File: src/modules/profile/pages/profileSettings/components/UserNotifications.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Agill-Sheron) [reply src/modules/profile/pages/profileSettings/components/UserNotifications.tsx:27]: There was no skeleton in this instance I think

**Review by Agill-Sheron — APPROVED** [review]

File: src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx [hunk 6]
```diff
@@ -1,49 +1,195 @@
-import { Input } from '@chakra-ui/react'
+import { FormControl, FormErrorMessage, Input, InputGroup, InputRightElement, Switch } from '@chakra-ui/react'
+import { debounce } from 'lodash'
+import { useCallback, useEffect, useState } from 'react'
 import { useTranslation } from 'react-i18next'
+import { BsFillCheckCircleFill, BsFillXCircleFill } from 'react-icons/bs'
```
Comment (sajald77) [inline src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx:5]: We've stopped using the icons from anything but react-icons/pi, which is the Phosphor Icons package. We're actively replacing other icons any touched portion of the code.

File: src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx [hunk 7]
```diff
@@ -1,49 +1,195 @@
...
-  if (!hasSelectedRewards) {
-    return null
+  /*
+   Set the email from the user to the funding form. We do this because the input field
+   is not shown if the user already has an email.
+  */
+  useEffect(() => {
+    if (user?.email) setTarget({ target: { name: 'email', value: user.email } })
+    else setTarget({ target: { name: 'email', value: '' } })
+  }, [user])
```
Comment (sajald77) [inline src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx:66]: there must be another export from the "useFundingFormAtom" hook, that is "setValue", which could be used for cases like these. setTarget was intended for use inside input field events, where values are found in event.target.value by default.

File: src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx [hunk 8]
```diff
@@ -1,49 +1,195 @@
...
+          setEmailValidationState(EMAIL_VALIDATION_STATE.FAILED)
+          setErrorstate({ key: 'email', value: error.message })
+        })
+      return Promise.resolve()
+    }, 1000),
+    [],
+  )
+
+  const renderEmailInputRightElement = () => {
+    if (userEmailIsAvailableLoading) return <Loader size="md"></Loader>
```
Comment (sajald77) [inline src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx:107]:
```suggestion
    if (userEmailIsAvailableLoading) return <Loader />
```

**Review by sajald77 — APPROVED** [review]
Great job stellios, @steliosrammos .

I added a few suggestions on improvements for consistency but overall looks solid.

File: src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx (hunk 7 above)
Comment (steliosrammos) [reply src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx:66]: I didn't see one but feel free to add it before merging the PR!

**Pull request merged**